orc8r/tools/fab/dev_utils.py

```diff
@@ -183,89 +184,103 @@ def create_tier_if_not_exists(
     """
     Get the hardware ID of a gateway running on Docker
 
     Args:
+        c: Fabric connection
         location_docker_compose: location of docker compose used to run FEG
         by default feg/gateway/docker
     Returns:
         Hardware snowflake from the VM
     """
-    with lcd('docker'), hide('output', 'running', 'warnings'), \
-            cd(location_docker_compose):
-        hardware_id = local(
-            'docker compose exec magmad bash -c "cat /etc/snowflake"',
-            capture=True,
-        )
-    return str(hardware_id)
+    host_data = vagrant.setup_env_vagrant(c, 'magma', force_provision=False)
+    with Connection(
+        host_data.get("host_string"),
+        connect_kwargs={"key_filename": host_data.get("key_filename")},
+    ) as c_agw:
+        with c_agw.cd(location_docker_compose):
+            hardware_id = c_agw.run(
+                'docker compose exec magmad bash -c "cat /etc/snowflake"',
+                hide=True,
+            ).stdout
+    return str(hardware_id).strip()
 
 
-def delete_gateway_certs_from_vagrant(vm_name: str):
+def delete_gateway_certs_from_vagrant(c: Connection, vm_name: str):
     """
     Delete certificates and gw_challenge of a gateway running on Vagrant VM
 
     Args:
+        c: Fabric connection
         vm_name: Name of the vagrant machine to use
     """
-    with settings(warn_only=True), hide('output', 'running', 'warnings'), \
-            cd('/var/opt/magma/certs'):
-        vagrant.setup_env_vagrant(vm_name)
-        sudo('rm gateway.*')
-        sudo('rm gw_challenge.key')
+    with c.cd(AGW_ROOT):
+        host_data = vagrant.setup_env_vagrant(c, vm_name, force_provision=False)
+        with Connection(
+            host=host_data.get("host_string"),
+            connect_kwargs={"key_filename": host_data.get("key_filename")},
+        ) as c_agw:
+            with c_agw.cd('/var/opt/magma/certs'):
+                c_agw.sudo('rm gateway.*', hide=True, warn=True)
+                c_agw.sudo('rm gw_challenge.key', hide=True, warn=True)
 
 
-def delete_gateway_certs_from_docker(location_docker_compose: str):
+def delete_gateway_certs_from_docker(c: Connection, location_docker_compose: str):
     """
         Delete certificates and gw_challenge of a gateway running on Docker
 
     Args:
+        c: Fabric connection
         location_docker_compose: location of docker compose used to run FEG
     """
-    print("delete_feg_certs is running on directory %s" % os.getcwd())
-
-    subprocess.check_call(
-        [
-            'docker compose exec magmad bash -c '
-            '"rm -f /var/opt/magma/certs/gateway.*"',
-        ],
-        shell=True,
-        cwd=location_docker_compose,
-    )
-
-    subprocess.check_call(
-        [
-            'docker compose exec magmad bash -c '
-            '"rm -f /var/opt/magma/certs/gw_challenge.key    "',
-        ],
-        shell=True,
-        cwd=location_docker_compose,
-    )
+    with c.cd(AGW_ROOT):
```

Comment:
why does the connection need to be switched/established here? It looks like this was not needed before

Reply:
Before, fabric did its magic and used whatever is written in `env.host` to connect to that host when useing `run`, `sudo`, `cd` etc. I.e. you'd have to go back through the code to find out where it actually connects. This is now much more clear by using context managers (although calling functions from within context managers gives kind of the same issue sometimes, I tried to solve that by naming the Connection variables within functions accordingly: `c` is the local host, `c_agw` is used for connecting to the dev VM, `c_vm` is used when the function may be called to connect to different vms, etc.).